numpy/core/src/npymath/halffloat.c

```diff
@@ -416,21 +418,26 @@ npy_uint16 npy_doublebits_to_halfbits(npy_uint64 d)
             npy_set_floatstatus_underflow();
         }
 #endif
-        d_sig >>= (1009 - d_exp);
+        /*
+         * Unlike floats, doubles has enough room to shift left to align
+         * significand. This changes the constant compared to normal branch.
+         */
+        assert(d_exp - 998 >= 0);
+        d_sig <<= (d_exp - 998);
```

Comment:
If you can shed any light in a comment on on where `998` comes from, that would be appreciated.